dojo/engagement/views.py

```diff
@@ -558,8 +558,11 @@ def import_scan_results(request, eid=None, pid=None):
 
             tt, t_created = Test_Type.objects.get_or_create(name=scan_type)
             # will save in development environment
-            environment, env_created = Development_Environment.objects.get_or_create(
-                name="Development")
+
+            environment_id = request.POST.get('environment', 'Default')
+            environment = Development_Environment.objects.get(id=environment_id)
+            environment, created = Development_Environment.objects.get_or_create(id=environment_id)
```

Comment:
the first one should be removed I guess?

Reply:
I did it intentionally to make the import fail if the `Environment` that was specified during the import does not exist, to be consistent with the API. I agree that it could be removed here as the selection is made from a pre-filled list in the UI, let me know what you think.

Reply:
I don't understand. If the environment doesn't exist, the first line will raise an exception. If it does exist, the 2nd statement won't do anything. So why have two statements?

Reply:
You're right that doesn't make any sense thanks for pointing that out. What I wanted to do initially is to make sure that the `Default` environment gets created if it doesn't exists to avoid breaking existing deployments (where the new `Default` environment wouldn't be provisioned by fixtures). I've modified a bit the logic, let me know what you think !

Reply:
I would do:
- UI: User selects from dropdown so we have to assume the environment exists under normal circumstances
- API: Environment can be freely supplied and not guaranteed to exist. I suggest to create the env if needed.

Reply:
* UI: Agree, make sense, I adapted the code
* API: I still think that we shouldn't automatically create the environment if it doesn't exist and let the user provision it separately. If we allow auto-create, users will end up having 2 or 3 time their environment with different cases or with typos `Development`, `development`, `dev`, and as there's no way to merge these environments it'll be nightmare-ish for companies with many teams to have something consistent over time.

Reply:
so then inboth cases this will suffice:
`environment = Development_Environment.objects.get(name=environment_name)`

Reply:
Yes you're right but if we do so for the API then it'll break existing integrations as people will be now forced to specify the `environment` in their POST. If we want to simply the code without breaking the existing integrations we could do:

```sh
environment, created = Development_Environment.objects.get_or_create(name=environment_name)`
```

Reply:
just leave it optional and leave the default at "Development" and we don't need the extra code. In the long term we better make the environment optional.

Reply:
Ok done !
I left the `Default` environment in the fixtures and in the unit-tests, this way we can maybe switch in a future release to this environment that make more sense than the `Development` one.